src/events/google_cloud.js

```diff
@@ -79,6 +87,132 @@ function bigQueryWrapper(wrappedFunction) {
+        }
+        let patchedCallback = callback;
+        try {
+            const pubsubProjectId = this.projectId;
+            const { event, startTime } = initializeEvent(
+                GOOGLE_CLOUD_TYPES.pubsub.name, pubsubProjectId, config.method
+            );
+            const requestFunctionThis = this;
+            const responsePromise = new Promise((resolve) => {
+                patchedCallback = (err, arg2, arg3) => {
+                    if ((!pubsubProjectId || pubsubProjectId === GOOGLE_CLOUD_TYPES
+                        .defaultProjectId) && !!requestFunctionThis.projectId) {
+                        event.getResource().setName(requestFunctionThis.projectId);
+                    }
+                    const callbackResponse = {};
+                    if (arg2) {
+                        if (config.method === 'publish') {
+                            const messageIds = arg2.messageIds ? arg2.messageIds : arg2;
+                            if (Array.isArray(messageIds) && messageIds.length) {
+                                callbackResponse.messageIds = messageIds;
+                                const [messageId] = messageIds;
+                                callbackResponse.messageId = messageId;
+                            }
+                        } else if (config.method === 'createSubscription') {
+                            callbackResponse.subscription = arg2;
+                        } else if (config.method === 'deleteSubscription') {
+                            if (config.reqOpts && config.reqOpts.subscription) {
+                                callbackResponse.subscription = getLastSplittedString(
+                                    config.reqOpts.subscription
+                                );
+                            }
+                        } else if (config.method === 'deleteTopic') {
+                            if (config.reqOpts && config.reqOpts.topic) {
+                                callbackResponse.topic = getLastSplittedString(
+                                    config.reqOpts.topic
+                                );
+                            }
+                        } else if (config.method === 'createTopic') {
+                            callbackResponse.topic = arg2;
+                        }
+                    }
+                    finalizeEvent(event, startTime, err, callbackResponse);
+                    resolve();
+                    if (callback) {
+                        if (arg3) {
+                            callback(err, arg2, ...arg3);
+                        } else {
+                            callback(err, arg2);
+                        }
+                    }
+                };
+            });
+            tracer.addEvent(event, responsePromise);
+        } catch (err) {
+            tracer.addException(err);
+        }
+        return original.apply(this, [config, patchedCallback]);
```

Comment:
are you sure this API has no option to not pass a callback and return a promise instead? please verify that.